packages/dd-trace/src/util.js

```diff
@@ -20,8 +20,20 @@ function isError (value) {
   return false
 }
 
+function parseRules (rules) {
+  if (Array.isArray(rules)) return rules
+  if (typeof rules !== 'string') return []
+  try {
+    const parsed = JSON.parse(rules)
+
+    if (!Array.isArray(parsed)) return []
```

Comment:
This could be validated in the sampler instead to that programmatic input is also checked.